packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -220,7 +221,12 @@ export function buildImportAnalysisPlugin(config: ResolvedConfig): Plugin {
           if (imports.length) {
             const s = new MagicString(code)
             for (let index = 0; index < imports.length; index++) {
-              const { s: start, e: end } = imports[index]
+              const { s: start, e: end, d: dynamicIndex } = imports[index]
+              // if dynamic import polyfill is used, rewrite the import to
+              // use the polyfilled function.
+              if (isPolyfillEnabled) {
+                s.overwrite(dynamicIndex, dynamicIndex + 6, `__import__`)
```

Comment:
**question:** What is `6`? It's a magic number in this context :thinking: 
Should we extract it into a variable/constant?

Reply:
6 is the number of letters in `import`, this is the same code that was removed before so it is ok to add it back as it was.

Reply:
Ah, so `'import'.length` would theoretically better here, but yeah, feel free to add it in a new PR or we could use non-squash-merge specially for this PR and use two commits :thinking:  
For readability and understandment, I would much prefer to use `'import'.length` here

Reply:
I think the context is enough here, if we add something in another PR maybe it could be a hint in the comment above like `'import'.length === 6'. I think this kind of patterns is not uncommon in the Vite code base though

Reply:
Honestly I don't think it's a magic number, as it's quite clear to see it's the length of `import` and giving the previous comment saying "rewrite the import". `'import'.length` would do but it's runtime, which to me, is not worth to bring the extra memory and CPU cost for the tiny DX that is not commonly touched anyway. I think it's ok to leave it as-is